torchgeo/datasets/nccm.py

```diff
@@ -0,0 +1,244 @@
+"""Northeastern China Crop Map Dataset."""
+
+import glob
+import os
+import shutil
+from collections.abc import Iterable
+from typing import Any, Callable, Optional, Union
+
+import matplotlib.pyplot as plt
+import torch
+from matplotlib.figure import Figure
+from rasterio.crs import CRS
+
+from .geo import RasterDataset
+from .utils import BoundingBox, download_url, extract_archive
+
+
+class NCCM(RasterDataset):
+    """The Northeastern China Crop Map Dataset.
+
+    Link: https://www.nature.com/articles/s41597-021-00827-9
+
+    This dataset produced annual 10-m crop maps of the
+    major crops (maize, soybean, and rice)
+    in Northeast China from 2017 to 2019, using hierarchial mapping strategies,
+    random forest classifiers, interpolated and
+    smoothed 10-day Sentinel-2 time series data and
+    optimized features from spectral, temporal and
+    textural characteristics of the land surface.
+    The resultant maps have high overall accuracies (OA)
+    based on ground truth data. The dataset contains information
+    specific to three years: 2017, 2018, 2019.
+
+    The dataset contains 4 classes:
+
+    0. paddy rice
+    1. maize
+    2. soybean
+    3. others
```

Comment:
Is there a background class too? Like does "others" include nodata pixels, other type of crops, non-cropland, etc? Or is there a separate class for nodata pixels?

Reply:
there is no background class. Others includes other crops, non-cropland etc. We had initially set a background class. But after we set it, all our plotting figures were only white and blank. It worked after we removed it.

Reply:
What classes exist in the original data? Are there no numbers larger than 3 or lower than 0?